Closing documents and closed documents limited interaction

codemoe opened the pull request:
Para que funcione:
1. Se debe correr "php artisan migrate"
2. Se debe relacionar los ids de esos registros en la tabla statuses en el archivo config/statuses
3. Se puede cambiar el status de un documento directamente en la pantalla de edición del documento

Added statuses config.
Added helpers to check document status.
Added messages for document closed.
Hide new comment on document closed or closed for comments.
Disabled comment actions on document closed.
Disabled annotations on document closed for comments.
Disabled annotation actions on document closed.
Added document closed legend on index and single document. Closes #99
Ordered documents by status (No status, opened, closed for comments, closed) Fixes #90
Added seed for statuses.
Added  unique index to label on statuses table.
Added insert for default statuses on migration.
Added document closed notification on action on annotator floating window.

Closes #1
closes #49

![captura de pantalla 2015-07-28 a las 18 33 00](https://cloud.githubusercontent.com/assets/4172905/8947250/4ea5910a-355e-11e5-8f9c-f0600ea54220.png)


![captura de pantalla 2015-07-28 a las 18 33 32](https://cloud.githubusercontent.com/assets/4172905/8947253/5293b382-355e-11e5-8333-d324d7f8984a.png)


Al intentar hacer una acción sobre un comentario en un documento cerrado...

![captura de pantalla 2015-07-28 a las 18 33 57](https://cloud.githubusercontent.com/assets/4172905/8947258/5e486060-355e-11e5-8cae-b53483bd1efd.png)

![captura de pantalla 2015-07-28 a las 18 59 52](https://cloud.githubusercontent.com/assets/4172905/8947267/70e9ba02-355e-11e5-81d0-ba64bca6a787.png)

codemoe commented:
@rodowi @urkonn Podrían probarlo para que nos aseguremos que todo funciona y que no se me pasó nada? Tanto para el documento cerrado para comentarios (que si permite dar like, etc.) como para el documento cerrado que no debe permitir ninguna interacción adicional

Para que funcione:
1. Deben existir los statuses en la tabla statuses que correspondan a documento abierto, documento cerrado y documento cerrado para comentarios.
2. Se debe relacionar los ids de esos registros en la tabla statuses en el archivo config/statuses
3. Se puede cambiar el status de un documento directamente en la pantalla de edición del documento

![captura de pantalla 2015-07-28 a las 19 35 28](https://cloud.githubusercontent.com/assets/4172905/8947374/d96f852e-355f-11e5-9d27-2e58cb2fa19a.png)

![captura de pantalla 2015-07-28 a las 19 35 37](https://cloud.githubusercontent.com/assets/4172905/8947376/dbce6ef2-355f-11e5-8be2-24fe4af47982.png)


cc @joecohens

urkonn commented:
@codemoe ¿hay que migrar la base de datos?

codemoe commented:
No hay que migrar pero si hay que insertar los statuses a la base de datos a mano....

urkonn commented:
funciona bien, lo único que no veo es el mensaje de alerta "Este documento se encuentra cerrado"

rodowi commented:
No podría ser una migración, el insertar valores a una tabla?

codemoe commented:
Agregué un par de cambios:

Added  unique index to label on statuses table.
Added insert for default statuses on migration.
Added document closed notification on action on annotator floating window.

@rodowi Hacer migrate debería agregar los 3 status a la tabla statuses. Lo que si hay que hacer es revisar que los ids de los status correspondan al archivo config/statuses

@urkonn Agregué una notificación más visible al intentar usar el annotator en un documento cerrado... No se si a esa notificación te referías....

![captura de pantalla 2015-07-29 a las 12 29 57](https://cloud.githubusercontent.com/assets/4172905/8964867/a2d99192-35ed-11e5-84a2-e6887b2f68a0.png)

urkonn commented:
@codemoe si, me refería a esa pero no la veo

codemoe commented:
@urkonn No ves la barra negra con el borde naranja en la parte superior de la pantalla o una notificación flotante roja en la esquina superior derecha?

Exactamente que interacción estas haciendo en donde? Son notificaciones diferentes si se usan los botones de acción del sidebar o si se usan los botones de acción en la ventana flotante sobre una anotación dentro del contenido...

¿El documento que estas probando está cerrado o cerrado para comentarios?

urkonn commented:
@codemoe ah ok, ya veo las dos

urkonn commented:
@codemoe esto va al 2.1

codemoe commented:
@urkonn Creo que a este ya no le falta nada más e incluye el fix para guardar las categorías... creo que si valdría la pena meterlo al 2.0

pero como ustedes vean...

urkonn commented:
el 2.1 puede liberarse en estos días

codemoe commented:
@urkonn @rodowi Ya está lista la limpieza y el rebase de este PR para que puedan revisarlo y probarlo...

urkonn commented:
gracias @codemoe

urkonn commented:
Me permite hacer comentarios en las anotaciones de un documento cerrado a comentarios
![selection_068](https://cloud.githubusercontent.com/assets/37691/9124240/efa25fe0-3c5a-11e5-8233-3e8272453805.png)

rodowi commented:
¿qué pasa cuando le das "Enviar"?

codemoe commented:
@urkonn Lo reviso para que tampoco se pueda hacer anotaciones...

El resto de las interacciones? algo más que hayas detectado?

urkonn commented:
al hacer refresh ya no aparece la anotación ni el comentario nuevo

urkonn commented:
@codemoe sigo probando

urkonn commented:
cerré este PR pero abrí los issues que están pendientes

rodowi commented:
@urkonn puedes hacer referencia a los issues que abre este PR?

urkonn commented:
abre #177 y #178
